tprs.go

```diff
@@ -164,3 +165,42 @@ func (t *ThirdPartyResources) List(ctx context.Context, resource, namespace stri
 	url := t.c.urlFor(t.apiGroup, t.apiVersion, namespace, resource, "")
 	return t.c.get(ctx, jsonCodec, url, resp)
 }
+
+type ThirdPartyResourcesWatcher struct {
+	watcher *jsonWatcher
+}
+
+func (w *ThirdPartyResourcesWatcher) Next(resp interface{}) (*JSONEvent, error) {
```

Comment:
Instead of returning `JSONEvent`, we can probably just convert it to a versioned.Event.

Reply:
I'll try with some unit tests to see if the same object can be used. Due to the fairly protobuf specific [RawExtension](https://github.com/ericchiang/k8s/blob/master/runtime/generated.pb.go#L81-L87) I couldn't manage to get the unmarshalling to work properly.